resources/certificate.rb

```diff
@@ -0,0 +1,28 @@
+resource_name :certificate
+default_action :install
+
+property :certfile, String
+property :cert_password, String
+property :keychain, String
+
+action_class do
+  def keychain
+    if property_is_set?(:keychain)
+      new_resource.keychain
+    else
+      ''
+    end
+  end
+end
+
+action :install do
+  test = SecurityCommand.new(new_resource.certfile, keychain)
+
+  execute 'unlock keychain' do
+    command [*test.unlock_keychain('vagrant')]
```

Comment:
Let's change `'vagrant'` to `node['macos']['admin_user']` attribute; just in case someones uses a different username in their environment.

Reply:
done